magento-research/pwa-studio
# Unit Tests for getQueryParameterValue and getSearchQueryParams

## Description



Do you want to hear something silly?

Unit tests for `getQueryParameter.js` were in fact present, they just didn't end in `.spec.js` and so they weren't being run 🤦‍♂️ .

I also noticed that `getSearchQueryParams` was a specific use case of `getQueryParameterValue` that didn't have tests and was only used by one file (`search.js`), so I replaced its usage there (in `search.js`) with the equivalent call to `getQueryParameterValue`. This lets us get rid of `util/getSearchParams.js` entirely 👍 .

## Related Issue





Closes #878 .

## Motivation and Context


Part of the ongoing effort to shore up our unit test coverage.

## How Has This Been Tested?




`yarn test`.

Also, for the change to the search page:
1. Navigate to the search page by executing a search
1. Refresh the browser page and verify that the search bar starts open
1. Execute a search and use the autocomplete dialog to select a category. Execute the search and verify that it is correct
1. Clear the category and verify that there are no errors

## Screenshots (if appropriate):

## Proposed Labels for Change Type/Package


TEST, BUG


`venia-concept`
## Checklist:


- [x] I have linked an issue to this PR.
- [x] I have indicated the change type and relevant package(s).
- [x] All CI checks are green (linting, build/deploy, etc).
- [x] At least one core contributor has approved this PR.
